src/Microsoft.Data.SqlClient/src/Microsoft/Data/SqlClient/TdsParser.cs

```diff
@@ -121,11 +123,17 @@ internal void TdsLogin(
             // length in bytes
             int length = TdsEnums.SQL2005_LOG_REC_FIXED_LEN;
 
-            string clientInterfaceName = TdsEnums.SQL_PROVIDER_NAME;
+            // Construct client interface name in format: Microsoft SqlClient - {OS}, {Platform} - {architecture}
+            // e.g. "Microsoft SqlClient - Microsoft Windows 10.0.26100, .NET 8.0.11 - X64"
+            string clientInterfaceName = new StringBuilder(TdsEnums.SQL_PROVIDER_NAME)
```

Comment:
This can be static class property. This wont need to be recomputed. It can be computed once and reused.